File: src/leetcode/python/MaximumAverageSubarrayI_643.py
class MaximumAverageSubarrayI_643(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        window = 0
        for i in k:
            window += nums[i]
        res = window
        for i in range(k, len(nums)):
            window += (nums[i] - nums[i - k])
            res = max(res, window)
        return float(res) / k

File: src/leetcode/java/MaximumAverageSubarrayI_643.java
package leetcode.java;

public class MaximumAverageSubarrayI_643 {
  public double findMaxAverage(int[] nums, int k) {
    int res = Integer.MIN_VALUE, window = 0, i = 0;
    for (; i < k; i++) window += nums[i];
    res = Math.max(res, window);
    for (; i < nums.length; i++) {
      window += (nums[i] - nums[i - k]);
      res = Math.max(res, window);
    }
    return (double) res / k;
  }
}
